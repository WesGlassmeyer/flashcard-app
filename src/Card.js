import React, { useState } from "react";
import Button from "./Button";

export default function Card({ frontSide, backSide, handleClick, handleNoClick }) {
  const [isFront, changeFace] = useState(true);

  function flipCard() {
    changeFace((oldState) => !oldState);
  }
  const text = isFront ? frontSide : backSide;
  const sideClass = isFront ? "front" : "back";
  const classList = `flash-card ${sideClass}`;
  return (
    <>
      <div className={classList} onClick={flipCard}>
        {text}
      </div>
      <Button text="Nailed It" style="green" handleClick={handleClick} />
      <Button text="Hell No" style="red" handleNoClick={handleNoClick} />
    </>
  );
}
