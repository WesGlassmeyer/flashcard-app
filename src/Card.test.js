import React from "react";
import ReactDOM from "react-dom";
import Card from "./Card";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

describe("Card component", () => {
  const props = {
    key: 1,
    frontSide: "sample text",
    backSide: "sample answer",
  };

  it("renders a card by default", () => {
    const wrapper = shallow(<Card />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders a card with props", () => {
    const wrapper = shallow(<Card {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Card />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
