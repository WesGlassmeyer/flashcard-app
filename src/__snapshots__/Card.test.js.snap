// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders a card by default 1`] = `
<Fragment>
  <div
    className="flash-card front"
    onClick={[Function]}
  />
  <Button
    style="green"
    text="Nailed It"
  />
  <Button
    style="red"
    text="Hell No"
  />
</Fragment>
`;

exports[`Card component renders a card with props 1`] = `
<Fragment>
  <div
    className="flash-card front"
    onClick={[Function]}
  >
    sample text
  </div>
  <Button
    style="green"
    text="Nailed It"
  />
  <Button
    style="red"
    text="Hell No"
  />
</Fragment>
`;
